lib/eslint.js

```diff
@@ -302,12 +302,12 @@ function modifyConfigsFromComments(filename, ast, config, reportingConfig, messa
     // apply environment configs
     Object.keys(commentConfig.env).forEach(function (name) {
         if (environments[name]) {
-            util.mergeConfigs(commentConfig, environments[name]);
+            assign(commentConfig, environments[name]);
```

Comment:
Cheeez I forgot to change it, I had in my head. (Spent too much time on the deep merge code lol)